### ISSUE TITLE
torch::tensor(scalar) behaves differently from at::tensor(scalar)

### ISSUE THREAD

**Issue by yf225** [pytorch/pytorch#28929]
After https://github.com/pytorch/pytorch/pull/28523 is merged, C++ `torch::tensor(scalar)` behaves the same as Python `torch.tensor(scalar)` and creates a 0-dim tensor. However, C++ `at::tensor(scalar)` still creates a 1-dim tensor, and it would be great to change the behavior of `at::tensor(scalar)` to create a 0-dim tensor.

**Comment by AlexandreFoley** [comment]
As far as i can tell, those constructor currently don't exist at all in the c++ api.
